croaring-sys/build.rs

```diff
@@ -5,12 +5,16 @@ use std::env;
 use std::path::PathBuf;
 
 fn main() {
-    cc::Build::new()
-        .flag_if_supported("-std=c11")
-        .flag_if_supported("-march=native")
-        .flag_if_supported("-O3")
-        .file("CRoaring/roaring.c")
-        .compile("libroaring.a");
+    let mut build = cc::Build::new();
+    build.file("CRoaring/roaring.c");
+
+    if let Ok(target_arch) = env::var("ROARING_ARCH") {
+        build.flag_if_supported(&format!("-march={}", target_arch));
+    } else {
+        build.flag_if_supported("-march=native");
```

Comment:
This is an unfortunate default, as it differs from Rust's. Using x86_64 as an example:

```
% rustc -Z unstable-options --target x86_64-unknown-linux-gnu --print target-spec-json
{
  "arch": "x86_64",
  "cpu": "x86-64",
  "data-layout": "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
  "dynamic-linking": true,
  "env": "gnu",
  "executables": true,
  "has-elf-tls": true,
  "has-rpath": true,
  "is-builtin": true,
  "linker-flavor": "gcc",
  "linker-is-gnu": true,
  "llvm-target": "x86_64-unknown-linux-gnu",
  "max-atomic-width": 64,
  "os": "linux",
  "position-independent-executables": true,
  "pre-link-args": {
    "gcc": [
      "-Wl,--as-needed",
      "-Wl,-z,noexecstack",
      "-m64"
    ]
  },
  "relro-level": "full",
  "stack-probes": true,
  "target-c-int-width": "32",
  "target-endian": "little",
  "target-family": "unix",
  "target-pointer-width": "64",
  "vendor": "unknown"
}
```

It seems like it would be better to rely on the environment variables to build a faster version, instead of the other way around. This matches Rust's `-C target-cpu=native`.